### ISSUE TITLE
Autocompletion for image:: and include:: files

### ISSUE THREAD

**Issue by rdmueller** [asciidoctor/asciidoctor-vscode#377]
The IntelliJ-Asciidoctor-Plugin manages to do an autocomplete on `image::` and `include::` macros to select a file.

This is quite helpful and avoids typos. I would like to see this feature here in the asciidoctor-vscode plugin :-)

Details:

* You start to type `include::` and press `ctrl`+`space`. 
* a drop down menu opens with a list of current folders and files to select from
* you select a folder and the next drop down opens, until you select a file
* you close your `include::` macro by adding `[]`

**Comment by d-koppenhagen** [comment]
@rdmueller should it really be necessary to press a keyboard shortcut like `ctrl` + `space`? I thing it would be even more easy and intuitive if the completion starts directly after typing `include::`.
This could work just like in the **PathIntellisense** Plugin (see here: https://github.com/ChristianKohler/PathIntellisense). WHat do you think about it?

**Comment by rdmueller** [comment]
sure! even better if you don't have to activate it 😀

**Comment by d-koppenhagen** [comment]
@rdmueller in addition, I added autocompletion also when using with defined variables. The completion will take tthe he pathes defined in the varaible selected from the completion list and starts to search for child elements inside the directory

**Comment by rdmueller** [comment]
wow! sounds perfect!
And do I see a PR there? awesome!

**Comment by d-koppenhagen** [comment]
Yes. There is still a small problem with just a specific case I linked a stackoverflow issue. Currently I have no idea how to solve it

**Comment by eiswind** [comment]
is there any progress on this? atm it seems a little broken to me.

**Comment by Mogztter** [comment]
> is there any progress on this? atm it seems a little broken to me.

Could you please be more specific?
Maybe you are referring to the fact that built-in attributes are included in the auto-completion?

This has been fixed in https://github.com/asciidoctor/asciidoctor-vscode/pull/529 and will be available in the next release.
Here's what I get using the master branch (next release):

![autocomplete](https://user-images.githubusercontent.com/333276/157022144-f0f3e587-8994-4910-af49-7d62bf27bc62.gif)

**Comment by Mogztter** [comment]
@eiswind Thanks that's really valuable. I think we should open a new issue to track these two feature enhancements. 

> I found that the completion seems to resolve the files and folders to the root of the project, which makes it almost useless

I don't think we are using the path before the cursor but we definitely should. In your case `{srcdir}/003_spring_boot/` is ignored and you will get the same suggestions no matter where in the target path your cursor is.

> And in my case i use an attribute to resolve the source folder, which seems to cause even more confusion.

It seems that we are trying to resolve the attribute but I guess `srcdir` is defined in another file?

https://github.com/asciidoctor/asciidoctor-vscode/blob/e96ac0b7f40c7f85c6fba3bf08db7f6d4aab3e36/src/providers/asciidoc.provider.ts#L45-L51

> Here you can see the folders from the project root:

I believe this is intended but I think we should see first files available within the current directory.

**Comment by eiswind** [comment]
The attribute is defined in the same file. Please let me know if I can provide assistance.

-------- Original-Nachricht --------

**Comment by Mogztter** [comment]
> The attribute is defined in the same file. You can see that in the first screenshot.

Oh I see... that does not work because the current implementation does not handle `ifndef` macro 😞 

> Plus I think that relative includes are always resolved to the directory the file lives in, but I am not completely sure about that.
Please let me know if I can provide assistance.

I will take a closer look, thanks for your feedback that's really valuable 👍🏻

**Comment by ggrossetie** [comment]
I've made quite a few improvements to autocompletion (next major version 4.0). Please open a new issue if there are still points that need improving.
